### ISSUE TITLE
[Feature Request] Right to left languages support

### ISSUE THREAD

**Issue by ma-sadeghi** [zadam/trilium#623]
It would be great if the editor natively supports right-to-left languages like Arabic and Farsi.

**Comment by zadam** [comment]
This will come in the next release (0.35) since CKEditor [just merged support for RTL](https://github.com/ckeditor/ckeditor5/issues/1887).

**Comment by fla15** [comment]
when 0.35 will be released?

**Comment by zadam** [comment]
0.35-beta has been already released. It contains the new CKEditor which should support RTL but I did not test RTL specifically (I don't really know how anyway). Feedback is welcome.

**Comment by fla15** [comment]
doesn't work. 
![0 35 0-beta](https://user-images.githubusercontent.com/30726939/64745276-130bee80-d4bc-11e9-9450-ffe18d9decba.png)
also the font is not standard.

**Comment by ma-sadeghi** [comment]
@zadam Any updates? Thanks

**Comment by zadam** [comment]
Not really, I guess this is one of those "help needed" issues since I have little experience with RTL. Based on CKEditor documentation it should work somehow automatically if the user has RTL environment (which I don't).

**Comment by BigGitWorld** [comment]
Please support RTL regardless of the user environment.
I'm using Ubuntu and need to RTL support for some my own notes.
I already use the gedit editor and it supports RTL.
You can also refer to the following link from the Notable:
[Notable-RTL-Support](https://github.com/notable/notable/issues/248)

**Comment by Hashemian01** [comment]
Any Updates?

**Comment by MAX10541** [comment]
@zadam there seems to be some partial, easy fixes for this. For example: using [CKEditor5 Text Part Language](https://ckeditor.com/docs/ckeditor5/latest/features/language.html), and [this plugin](https://github.com/abedi-ir/ckeditor5-direction/).  
I haven't tested them but I can give a hand if you decide to try them.

**Comment by hosseinvand** [comment]
Is there any updates? I know a ton of people with rtl language needing this feature

**Comment by smhhf23** [comment]
Is there any updates? to solve this problem ?

**Comment by yshalsager** [comment]
@zadam 

> Based on CKEditor documentation it should work somehow automatically if the user has RTL environment.

I was just trying, seems it won't change [contentLanguageDirection](https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html#setting-the-language-of-the-content) automatically as @matty0blake said.

> the solution will be to add the icon tool to the tool bar. i did try to replace it with a CKEditor that i build it with having the direction changing icon on the tool bar.

This sounds a good and easy solution. I have tried to do it but couldn't manage to get all the pieces together since I am not familiar with the API.

**Comment by eliandoran** [comment]
@ma-sadeghi , @yshalsager , @smhhf23 , @hosseinvand , @Hashemian01 , @BigGitWorld , @fla15 , I would be willing to have a look at this for the TriliumNext work. However I don’t have any experience with RTL as user, would you be willing to help test this feature?

**Comment by ma-sadeghi** [comment]
Sure, I can volunteer.

**Comment by eliandoran** [comment]
@ma-sadeghi , sounds good, I'll have a go at adding support for writing RTL in the editor.
I would also appreciate any help in adding support for a RTL language within Trilium, by helping with the translations (~1500 messages).

**Comment by ma-sadeghi** [comment]
Sure, it'd be my pleasure!

**Comment by eliandoran** [comment]
Tracking the development of RTL for text notes here: https://github.com/TriliumNext/Notes/issues/97

**Comment by eliandoran** [comment]
Preliminary support for RTL added in https://github.com/TriliumNext/Notes/pull/1319.
